### DISCUSSION TITLE
Color Connection

### DISCUSSION THREAD

**Discussion by matrix4767** [lllyasviel/ControlNet#74] [Ideas]
What tech would be needed for a color model that:
-Identifies an image's colors and even proper shade of said colors
-Can assign colors to the output based on the initial image's colors of eyes/hair/clothes/objects. So that a teal haired, orange eyed, white  pants and green shirt wearing person would guarantee those colors and their shades to the generated image.

**Comment by Njasa2k** [comment]
isn't this what lllyasviel's other project does? https://github.com/lllyasviel/style2paints

**Reply by matrix4767** [reply]
Yes. Perhaps that as an extension that would lock colors on generation.

**Comment by matrix4767** [comment]
This might be relevant? Color palette can be done somehow
https://github.com/damo-vilab/composer
https://arxiv.org/pdf/2302.09778.pdf

**Reply by Njasa2k** [reply]
> This might be relevant? Color palette can be done somehow https://github.com/damo-vilab/composer https://arxiv.org/pdf/2302.09778.pdf

Unfortunately that requires 28GB of VRAM as what the author told me, he says he can optimize it enough to work on consumer graphic cards but we will see how that goes.

**Reply by matrix4767** [reply]
What of _just_ adapting the palette part and not the whole composer as a contronet variant?

**Comment by geroldmeisinger** [comment]
controlnet color https://huggingface.co/thibaud/controlnet-sd21#color
